src/test/java/org/mockito/internal/configuration/plugins/PluginLoaderTest.java

```diff
@@ -0,0 +1,87 @@
+package org.mockito.internal.configuration.plugins;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.io.IOUtil;
+import org.mockitoutil.ClassLoaders;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test PluginLoader. This test check that loaded plugin can be assigned to fields of classes what loaded by
+ * application class loader
+ */
+public class PluginLoaderTest {
+
+    public static final String PLUGIN_PATH = "mockito-extensions/org.mockito.internal.configuration.plugins.PluginInterface";
+    private PluginLoader pluginLoader;
+
+    @Before
+    public void setUp() {
+        pluginLoader = new PluginLoader(new DefaultPluginSwitch());
+    }
+
+    @Test
+    public void should_load_custom_plugin_without_context_classloader() {
+
+        Object pluginInterface = loadPlugin();
+
+        assertThatLoadedPluginAssignable(pluginInterface);
+    }
+
+
+    @Test
+    public void should_load_custom_plugin_with_custom_context_classloader() throws Exception {
+
+        ClassLoader classLoader = ClassLoaders.inMemoryClassLoader().withParent(getParentClassLoader())
+                                          .withClassDefinition("", readResource()).build();
```

Comment:
I understand the idea, but the test pass with or without `.withClassDefinition("", readResource())`.

The `ClassLoaders` is just a basic tool that wasn't designed to support something else but class bytes. It may be something to improve. Anyway it's just coincidental that file is supported.

Anyway for better readability it would have been clearer to write it like that, and then it's possible to discard `readResource()` :

```java
ClassLoader classLoader = ClassLoaders.inMemoryClassLoader()
                                      .withParent(getParentClassLoader())
                                      .withClassDefinition("mockito-extensions/org.mockito.internal.configuration.plugins.PluginInterface", 
                                                           PluginImpl.class.getName().getBytes("UTF-8"))
                                      .build();
```